src/processor/operator/shortest_path.cpp

```diff
@@ -0,0 +1,100 @@
+#include "processor/operator/shortest_path.h"
+
+#include "iostream"
+
+using namespace std;
+
+namespace kuzu {
+namespace processor {
+
+bool ShortestPathAdjList::getNextTuplesInternal() {
+    while (true) {
+        while (!bfsQueue.empty()) {
+            auto queueFrontNodeOffset = bfsQueue.front();
+            auto bfsFrontNodeState = bfsVisitedNodesMap[queueFrontNodeOffset];
+            if (bfsFrontNodeState->bfsLevel >= lowerBound &&
+                bfsFrontNodeState->bfsLevel <= upperBound && bfsFrontNodeState->hasBeenOutput) {
+                for (auto i = destValueVector->state->getPositionOfCurrIdx();
```

Comment:
I don't understand the for loop here. Why are you looping over dest value vector?

Reply:
Yeah I had done this for keeping it generic like if there are multiple destination nodes for same source.
But we kept `Flatten` so only 1 source and 1 destination should be present at any point.

I've removed the for loop and added code for only 1 destination node now.